### PR TITLE
Implementing channel-closing.

### PR DESCRIPTION
**Pull request by cprov**
/dev/api/snaps/<snap-id>/close now returns currently closed channels and channel_maps across snap architectures (status).

LP: #1615764

### CODE REVIEW

File: integration_tests/test_store_close.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import uuid
+
+from testtools.matchers import (
+    FileExists,
+)
```
Comment (sergiusens) [inline integration_tests/test_store_close.py]: pet peeve, can this be a one liner?

File: integration_tests/test_store_close.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+import integration_tests
+
+
+class ChannelClosingTestCase(integration_tests.StoreTestCase):
+
+    project_dir = 'basic'
+
+    def setUp(self):
+        super().setUp()
```
Comment (sergiusens) [inline integration_tests/test_store_close.py]: please remove `def setUp` completely, seems like a left over.

File: snapcraft/_store.py [hunk 3]
```diff
@@ -379,6 +379,55 @@ def release(snap_name, revision, release_channels):
     print(tabulated_channels)
 
 
+def _tabulated_status(status):
```
Comment (sergiusens) [inline snapcraft/_store.py]: :+1:

File: snapcraft/tests/fake_servers.py [hunk 4]
```diff
@@ -426,6 +430,59 @@ def _handle_sign_build_request(self):
...
+        payload = json.loads(string_data)
+        channels = payload['channels']
+
+        if channels == ['invalid']:
+            self.send_response(400)
+            self.send_header('Content-Type', 'application/json')
+            error = {
+                'error_list': [
+                    {'code': 'invalid-field',
+                     'message': ('The \'channels\' field content is not '
```
Comment (sergiusens) [inline snapcraft/tests/fake_servers.py]: Not related to the mock, but would be nice if the store returned a list of valid channels to use.

File: snapcraft/_store.py [hunk 5]
```diff
@@ -379,6 +379,55 @@ def release(snap_name, revision, release_channels):
...
+    store = storeapi.StoreClient()
+
+    with _requires_login():
+        info = store.get_account_information()
+
+    try:
+        snap_id = info['snaps'][snap_series][snap_name]['snap-id']
+    except KeyError:
+        raise RuntimeError(
+            'Your account lacks permission to close channel for this snap. '
```
Comment (sergiusens) [inline snapcraft/_store.py]: either _to close channels_ or _to close a channel_

File: integration_tests/test_store_close.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    def test_missing_permission(self):
+        self.addCleanup(self.logout)
+        self.login()
+        expected = (
+            'Make sure the logged in account has upload permissions on '
+            "'missing' in series '16'.")
+        status = self.close('missing', 'beta', expected=expected)
+        self.assertEqual(1, status)
+
+    def test_simple(self):
```
Comment (elopio) [inline integration_tests/test_store_close.py]: A slightly better name would be something like test_close_channel

**Comment by elopio** [comment]
I did a small exploratory on this branch and couldn't find any problems. It looks like it's only missing a few details commented by Sergio. Thank you @cprov.

File: integration_tests/test_store_close.py (hunk 2 above)
Comment (cprov) [inline integration_tests/test_store_close.py]: yup, fixed.

File: snapcraft/tests/fake_servers.py (hunk 4 above)
Comment (cprov) [inline snapcraft/tests/fake_servers.py]: That's a good point, I will proposed a fix in the store for push, release & close (similar errors) and submit a new PR here.

File: integration_tests/test_store_close.py (hunk 6 above)
Comment (cprov) [inline integration_tests/test_store_close.py]: sure, fixed.

**Pull request merged**